### PR TITLE
Functional tests for UIKit Extensions

### PR DESCRIPTION
**Pull request by skyylex**
Related to the https://github.com/neilpa/Rex/issues/16.

@neilpa, please review.

### CODE REVIEW

File: Tests/UIKit/UIBarButtonItemTests.swift [hunk 1]
```diff
@@ -28,5 +28,17 @@ class UIBarButtonItemTests: XCTestCase {
         }
         barButtonItem.rex_action <~ SignalProducer(value: CocoaAction(action, input: ()))
      }
+    
+    func testEnabledProperty() {
+        let barButtonItem = UIBarButtonItem()
+        let (pipeSignal, observer) = Signal<Bool, NoError>.pipe()
+        barButtonItem.rex_enabled <~ SignalProducer(signal: pipeSignal)
+        barButtonItem.enabled = true
```
Comment (neilpa) [inline Tests/UIKit/UIBarButtonItemTests.swift]: You shouldn't set the enabled state explicitly (at least not after binding)

File: Tests/UIKit/UIButtonTests.swift [hunk 2]
```diff
@@ -10,6 +10,17 @@ import ReactiveCocoa
 import UIKit
 import XCTest
 
+public extension UIButton {
```
Comment (neilpa) [inline Tests/UIKit/UIButtonTests.swift]: These aren't used

File: Tests/UIKit/UIBarButtonItemTests.swift [hunk 3]
```diff
@@ -28,5 +28,17 @@ class UIBarButtonItemTests: XCTestCase {
...
+        
+        observer.sendNext(false)
+        XCTAssert(barButtonItem.enabled == false, "#1 change of the enabled state failed [UIBarButtonItem]")
```
Comment (neilpa) [inline Tests/UIKit/UIBarButtonItemTests.swift]: You should use `XCTAssertFalase(barButtonItem.enabled)`.

File: Tests/UIKit/UIBarButtonItemTests.swift [hunk 4]
```diff
@@ -28,5 +28,17 @@ class UIBarButtonItemTests: XCTestCase {
...
+        observer.sendNext(true)
+        XCTAssert(barButtonItem.enabled == true, "#2 change of the enabled state failed [UIBarButtonItem]")
```
Comment (neilpa) [inline Tests/UIKit/UIBarButtonItemTests.swift]: `XCTAssertTrue`

Comment (neilpa) [inline Tests/UIKit/UIBarButtonItemTests.swift]: Can you also remove these messages. They just duplicate information that xcode will already display by default.

**Comment by neilpa** [comment]
Can you remove all of the messages that you've added to the `XCTAssert` calls. Generally Xcode already does a good job displaying what failed in line and these will just clutter those.

File: Tests/UIKit/UIButtonTests.swift [hunk 5]
```diff
@@ -44,5 +55,45 @@ class UIButtonTests: XCTestCase {
...
+        let firstTitle = "First title"
+        let secondTitle = "Second title"
+        let button = UIButton(frame: CGRectZero)
+        let (pipeSignal, observer) = Signal<String, NoError>.pipe()
+        button.rex_title <~ SignalProducer(signal: pipeSignal)
+        button.setTitle("", forState: .Selected)
+        button.setTitle("", forState: .Highlighted)
+        
+        observer.sendNext(firstTitle)
+        XCTAssert(button.titleForState(.Normal) == firstTitle, "#1 change of the title for .Normal state failed [UIButton]")
```
Comment (neilpa) [inline Tests/UIKit/UIButtonTests.swift]: Can you use `XCTAssertEqual` for these comparisons and throughout

**Comment by neilpa** [comment]
Can you also update to `XCTAssertEqual`, `XCTAssertTrue` and `XCTAssertFalse` where appropriate.

**Comment by skyylex** [comment]
@neilpa, fixed.

**Comment by neilpa** [comment]
Awesome! Thanks for the PR :bow:

**Pull request merged**